### PR TITLE
Back to getters and setters, Edited checkStringIsValid

### PR DESCRIPTION
**Pull request by TryExceptElse**
revising getters and setters (made pull request)

### CODE REVIEW

File: src/main/java/io/github/tryexceptelse/jdex/be/IRolodex.java [hunk 1]
```diff
@@ -16,26 +17,29 @@
...
+     * @param last:       expect LastName object representing the last name of the contact
+     * @param email:      expects EmailAddress object representing the email address of the contact
+     * @param streetAddr: expects EmailAddress object represnting the email address of the contact
+     * @param phone:      expects PhoneNumber object representing the phone number of the contact
+     * @param notes:      expects ContactNotes object representing notes for the contact
      */
-    Contact addContact(HashMap<String, ContactEntry> newEntries);
+
+    public Contact addContact(FirstName first,
+                              LastName last,
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/IRolodex.java]: Constructor methods of Rolodexshould initialize the contacts ArrayList, or the other methods will not have anything to operate on

File: src/main/java/io/github/tryexceptelse/jdex/be/Rolodex.java [hunk 2]
```diff
@@ -26,7 +22,8 @@
      * Default constructor for Rolodex.
      * Should use default file location
      */
-    public Rolodex() {
+    public Rolodex()
+    {
         // skeleton placeholder.
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/Rolodex.java]: Constructors of Rolodex should initialize the contacts ArrayList or the other methods will have nothing to operate on

File: src/main/java/io/github/tryexceptelse/jdex/be/Contact.java [hunk 3]
```diff
@@ -1,52 +1,154 @@
...
+     * @param email:      expects EmailAddress object representing the email address of the contact
+     * @param streetAddr: expects EmailAddress object represnting the email address of the contact
+     * @param phone:      expects PhoneNumber object representing the phone number of the contact
+     * @param notes:      expects ContactNotes object representing notes for the contact
      */
-    public Contact(HashMap<String, ContactEntry> entries)
+    public Contact(FirstName first,
+                   LastName last,
+                   EmailAddress email,
+                   StreetAddress streetAddr,
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/Contact.java:42]: These variables need to be stored in the created contact

File: src/main/java/io/github/tryexceptelse/jdex/be/Rolodex.java [hunk 4]
```diff
@@ -60,38 +58,51 @@ public void setContacts(ArrayList<Contact> newContacts) {
...
+        // Checking to make sure all passed fields are valid. The methods are defined in their own classes
+        for (ContactEntry entry : new ContactEntry[]{first, last, email, streetAddr, phone, notes})
+        {
+            if (!entry.checkHasValidString())
+                throw new InvalidObjectException(entry + "was invalid. Could not add to contacts");
+        }
+        //adding contact to the contatcs ArrayList
+        contacts.add(new Contact(first, last, email, streetAddr, phone, notes));
+        return new Contact();
     }
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/Rolodex.java:101]: addContact, as written, is returning a totally different contact object than the one created using the passed entries.

File: src/main/java/io/github/tryexceptelse/jdex/be/Contact.java [hunk 5]
```diff
@@ -1,52 +1,191 @@
...
+        this.phone=phone;
+        this.notes=notes;
+    }
+    /**
+     * Calls the checkHasValidString method on each of the parameters of the object
+     */
+    public boolean checkParametersAreValid()
+    {
+        // Checking to make sure all passed fields are valid. The methods are defined in their own classes
+        if (last.checkHasValidString())
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/Contact.java]: checkHasValidString returns true if the string is valid, and judging by its use elsewhere, this method is also meant to return true if all entries are valid. This method as written however returns false if any of the ContactEntry objects are valid.

Also, this whole thing could be expressed in a for-x-in-y loop, and be a lot shorter

File: src/main/java/io/github/tryexceptelse/jdex/be/entries/LastName.java [hunk 6]
```diff
@@ -19,4 +19,27 @@ public LastName() {
...
+     * @return: will return False if string is empty, or just contains ' ' chars.
+     */
+    public static boolean checkStringIsValid(String entryString)
+    {
+        if(!ContactEntry.checkStringIsValid(entryString))
+           return false;
+        else if (entryString==null)
+            return false;
+        //iterates through each char of the String and checks if it is a ' '. If one of them is not, returns true
+        String str = new String(entryString);
```
Comment (TryExceptElse) [inline src/main/java/io/github/tryexceptelse/jdex/be/entries/LastName.java]: why is a new string created here which is identical to entryString? and why is a String constructor used?

**Comment by TryExceptElse** [comment]
Sadly, everything does not quite work yet

**Comment by Entrer1** [comment]
It's all your fault that things are broken

**Pull request merged**